archiver.go

```diff
@@ -1,33 +1,172 @@
 package debos
 
 import (
-	"log"
+	"fmt"
 	"os"
+	"path/filepath"
+	"strings"
 )
 
-func TarOptions(compression string) string {
+type ArchiveType int
+
+// Supported types
+const (
+	_ ArchiveType = iota // Guess archive type from file extension
+	Tar
+	Zip
+	Deb
+)
+
+type Archive struct {
+	File        string // Path to archive file
+	Type        ArchiveType
+	compression string
+}
+
+type Archiver interface {
+	CommandUnpack(destination string, options ...string) []string
+	GetType() ArchiveType
+	SetCompression(compression string) error
+	Unpack(destination string, options ...string) error
```

Comment:
Allowing the caller to add extra options to the unpack command breaks your abstraction. There is no reason for the user to know underneath just unix commands are being called especailly as this may change at some point.